packaging/linux/post_install.sh

```diff
@@ -9,33 +9,74 @@ set -u
 
 rootmount="/keybase"
 krbin="/usr/bin/keybase-redirector"
+rootConfigFile="/etc/keybase/config.json"
+disableConfigKey="disable-root-redirector"
 
 vardirDeprecated="/var/lib/keybase"
 khuserDeprecated="keybasehelper"
 khbinDeprecated="/usr/bin/keybase-mount-helper"
 optDeprecated="/opt/keybase/mount-readme"
 
-chown root:root "$krbin"
-chmod 4755 "$krbin"
+redirector_enabled() {
+  disableRedirector="false"
+  if [ -r "$rootConfigFile" ] ; then
+    if keybase --standalone -c "$rootConfigFile" config get -d "$disableConfigKey" &> /dev/null ; then
+      disableRedirector="$(keybase --standalone -c "$rootConfigFile" config get -d "$disableConfigKey" 2> /dev/null)"
+    fi
+  fi
+  [ "$disableRedirector" != "true" ]
+}
 
 make_mountpoint() {
-  if ! mountpoint "$rootmount" &> /dev/null; then
-    mkdir -p "$rootmount"
-    chown root:root "$rootmount"
-    chmod 755 "$rootmount"
+  if redirector_enabled ; then
+    if ! mountpoint "$rootmount" &> /dev/null; then
+      mkdir -p "$rootmount"
+      chown root:root "$rootmount"
+      chmod 755 "$rootmount"
+    fi
   fi
 }
 
 run_redirector() {
   make_mountpoint
 
-  logdir="${XDG_CACHE_HOME:-$HOME/.cache}/keybase"
-  mkdir -p "$logdir"
-  nohup "$krbin" "$rootmount" >> "$logdir/keybase.redirector.log" 2>&1 &
+  # Only start the root redirector if it hasn't been explicitly disabled.
+  if redirector_enabled ; then
+    logdir="${XDG_CACHE_HOME:-$HOME/.cache}/keybase"
+    mkdir -p "$logdir"
+    echo Starting root redirector at $rootmount.
+    nohup "$krbin" "$rootmount" >> "$logdir/keybase.redirector.log" 2>&1 &
+    t=5
+    while ! mountpoint "$rootmount" &> /dev/null; do
+        sleep 1
+        t=$[t-1]
+        if [ $t -eq 0 ]; then
+            echo "Redirector hasn't started yet."
+            break
+        fi
+    done
+  elif killall `basename "$krbin"` &> /dev/null ; then
+    echo "Killing root redirector"
+  fi
 }
 
-currlink="$(readlink -m "$rootmount")"
+if redirector_enabled ; then
+  chown root:root "$krbin"
+  chmod 4755 "$krbin"
+else
+  # Turn off suid if root has been turned off.
+  chmod 0755 "$krbin"
```

Comment:
It might be possible to do `chmod a-s "$krbin"` here. I think it's a wash in terms of clarity, but if for some reason the admin had removed execute permissions from the binary or something weird like that, I think that command would avoid re-adding them.

Reply:
Done thanks.